challenge_2/go/erocs/find_singleton.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+)
+
+func FindSingleton(int_array []int) (found int) {
+	singles := make(map[int]int)
+	repeats := make(map[int]int)
+	for _, n := range int_array {
+		if _, ok := repeats[n]; ok {
+			repeats[n]++
+		} else if _, ok := singles[n]; ok {
+			delete(singles, n)
+			repeats[n] = 2
+		} else {
+			singles[n] = 1
+		}
+	}
+	for k, _ := range singles {
+		found = k
```

Comment:
Might as well exit the for loop when you find one since you'll just be reassigning `found` each time you find one and it doesn't look like it matters what one you return.